pinot-core/src/test/java/com/linkedin/pinot/core/realtime/RealtimeSegmentTest.java

```diff
@@ -97,7 +97,8 @@ public static void before() throws Exception {
     when(segmentDataManager.getInvertedIndexColumns()).thenReturn(invertedIdxCols);
     when(segmentDataManager.getNoDictionaryColumns()).thenReturn(new ArrayList<String>());
     when(segmentDataManager.getSchema()).thenReturn(schema);
-    when(segmentDataManager.getMemoryManager()).thenReturn(new DirectMemoryManager("noSegment"));
+    when(segmentDataManager.getMemoryManager()).thenReturn(new DirectMemoryManager("noSegment",
```

Comment:
Do tests really need to send a ServerMetrics object? Would it be better to just pass null, or have another constructor without this parameter, so that clients that do need metrics (eg test) don't need to worry about it?

Reply:
A toss-up. I could special-case it inside the code for tests, like I did for segment name.  agreed i could have been consistent with segment name and metrics (the metrics case came in automagically with IDE :-). I can undo the test changes and special-case it if metrics object is null.